backend/migrations/2023-07-15-152628_squashed_setup/up.sql

```diff
@@ -0,0 +1,461 @@
+    'cuttings',
+    'layering',
+    'spores',
+    'seed'
+);
+
+CREATE TYPE quality AS ENUM (
+    'organic',
+    'not organic',
+    'unknown'
+);
+
+CREATE TYPE quantity AS ENUM (
+    'nothing',
+    'not enough',
+    'enough',
+    'more than enough'
+);
+
+CREATE TYPE relation_type AS ENUM (
+    'companion',
+    'neutral',
+    'antagonist'
+);
+
+CREATE TYPE shade AS ENUM (
+    'no shade',
+    'light shade',
+    'partial shade',
+    'permanent shade',
+    'permanent deep shade'
+);
+
+CREATE TYPE soil_ph AS ENUM (
+    'very acid',
+    'acid',
+    'neutral',
+    'alkaline',
+    'very alkaline'
+);
+
+CREATE TYPE soil_texture AS ENUM (
+    'sandy',
+    'loamy',
+    'clay',
+    'heavy clay'
+);
+
+CREATE TYPE soil_water_retention AS ENUM (
+    'well drained',
+    'moist',
+    'wet'
+);
+
+CREATE TYPE water_requirement AS ENUM (
+    'dry',
+    'moist',
+    'wet',
+    'water'
+);
+
+CREATE FUNCTION check_layer_type() RETURNS trigger
+    LANGUAGE plpgsql
+    AS $$
+BEGIN
+    IF (SELECT type FROM layers WHERE id = NEW.layer_id) != 'plants' THEN
+        RAISE EXCEPTION 'Layer type must be "plants"';
+    END IF;
+    RETURN NEW;
+END;
+$$;
+
+
+CREATE TABLE layers (
+    id integer NOT NULL,
+    map_id integer NOT NULL,
+    type layer_type NOT NULL,
+    name text NOT NULL,
+    is_alternative boolean NOT NULL
+);
+
+
+CREATE SEQUENCE layers_id_seq
+    AS integer
+    START WITH 1
+    INCREMENT BY 1
+    NO MINVALUE
+    NO MAXVALUE
+    CACHE 1;
+
+
+CREATE TABLE maps (
+    id integer NOT NULL,
+    name text NOT NULL,
+    creation_date date NOT NULL,
+    deletion_date date,
+    last_visit date,
+    is_inactive boolean NOT NULL,
+    zoom_factor smallint NOT NULL,
+    honors smallint NOT NULL,
+    visits smallint NOT NULL,
+    harvested smallint NOT NULL,
+    privacy privacy_options NOT NULL,
+    description text,
+    location geography(Point,4326),
+    owner_id uuid NOT NULL
+);
+
+CREATE SEQUENCE maps_id_seq
+    AS integer
+    START WITH 1
+    INCREMENT BY 1
+    NO MINVALUE
+    NO MAXVALUE
+    CACHE 1;
+
+CREATE TABLE plants (
+    id integer NOT NULL,
+    unique_name text NOT NULL,
+    common_name_en text[],
+    common_name_de text[],
+    family text,
+    genus text,
+    edible_uses_en text,
+    --medicinal_uses text,
+    material_uses_and_functions text,
+    botanic text,
+    --material_uses text,
+    functions text,
+    heat_zone smallint,
+    shade shade,
+    soil_ph soil_ph[],
+    soil_texture soil_texture[],
+    soil_water_retention soil_water_retention[],
+    --environmental_tolerances text[],
+    --native_geographical_range text,
+    --native_environment text,
+    ecosystem_niche text,
+    deciduous_or_evergreen deciduous_or_evergreen,
+    herbaceous_or_woody herbaceous_or_woody,
+    life_cycle life_cycle[],
+    growth_rate growth_rate[],
+    height plant_height,
+    fertility fertility[],
+    --flower_colour text,
+    flower_type flower_type,
```

Comment:
height will be used by me in the next few days, the import scraper for the mergedata.csv is already done. 
fertility & flower_type is used in `backend/src/schema.patch` and cant be removed. Is it important @ThorbenStaufer?

Reply:
> Is it important

Yes, the patch file is very important for our PostGis integration to work.
But it can simply be adapted or re-generated in your case, since fertility and flower_type are not crucial parts of it.

Reply:
since it also generates issue for the down.sql script, lets discuss it in another issue if and how we gonna re-generate it. The database still can be reset with `diesel database reset`

Reply:
`database reset` doesn't do anything for the issues that arise if we don't have a functioning `schema.patch`.

Reply:
sorry i missed an important part. Right now `diesel migration redo` throws `error applying hunk #1` but clears `schema.rs`. i need to change `schema.patch` im just not sure how yet

Reply:
The patch file itself is just a git diff file, that is read by diesel to make custom adaptations to the schema file.
There are two ways you could fix it:
1. manually update the patch file, so it can correctly detect the location in the schema file, that needs to be updated.
2. delete the old patch file, run diesel to get a filled schema file, manually make the changes in the schema file and then run `git diff -U6 > src/schema.patch` to let git generate a new diff file from your manual changes.

Reply:
Addendum to solution 1:
Manually updating the patch file is a bit tricky at first. Not only do you need to make sure that the code snippet inside the patch file perfectly matches the schema file how it would be generated by diesel without the patch file, but you also need to update the line count in the diff header. Without that diesel will not recognize it as a valid git diff.

Reply:
Addendum to solution 2:
For git to detect manual changes to the schema file, you need to temporarily remove it from git ignore and commit it to your local repository. It is important, that you first commit it and only after that make manual changes, since the git diff file would include the entire schema otherwise. Just don't forget to include it in gitignore again later and commit it's deletion.